sklearn/tests/test_random_projection.py

```diff
@@ -206,14 +206,7 @@ def test_sparse_random_matrix():
 ###############################################################################
 
 
-@pytest.mark.parametrize("density", [1.1, 0, -0.1])
-def test_sparse_random_projection_transformer_invalid_density(density):
-    for RandomProjection in all_SparseRandomProjection:
-        with pytest.raises(ValueError):
-            RandomProjection(density=density).fit(data)
-
-
-@pytest.mark.parametrize("n_components, fit_data", [("auto", [[0, 1, 2]]), (-10, data)])
+@pytest.mark.parametrize("n_components, fit_data", [("auto", [[0, 1, 2]])])
 def test_random_projection_transformer_invalid_input(n_components, fit_data):
     for RandomProjection in all_RandomProjection:
```

Comment:
Since they only take on one value, they don't need to be parameterized function arguments anymore and can just be local variables.
```suggestion
def test_random_projection_transformer_invalid_input():
        n_components = "auto"
        fit_data = [[0, 1, 2]]
        for RandomProjection in all_RandomProjection:
```